core-bundle/src/Resources/contao/classes/Versions.php

```diff
@@ -459,14 +459,19 @@ public function compare($blnReturnBuffer=false)
 							continue;
 						}
 
-						if (\is_array($arrFields[$k]))
-						{
-							// Detect binary fields using Doctrine's built-in types or Contao's BinaryStringType (see #3665)
-							$blnIsBinary = \in_array($arrFields[$k]['type'] ?? null, array(BinaryType::class, BlobType::class, Types::BINARY, Types::BLOB, BinaryStringType::NAME), true);
-						}
-						else
+						$blnIsBinary = false;
+
+						if (isset($arrFields[$k]))
 						{
-							$blnIsBinary = strncmp($arrFields[$k], 'binary(', 7) === 0 || strncmp($arrFields[$k], 'blob ', 5) === 0;
+							if (\is_array($arrFields[$k]))
+							{
+								// Detect binary fields using Doctrine's built-in types or Contao's BinaryStringType (see #3665)
+								$blnIsBinary = \in_array($arrFields[$k]['type'] ?? null, array(BinaryType::class, BlobType::class, Types::BINARY, Types::BLOB, BinaryStringType::NAME), true);
+							}
+							else
+							{
+								$blnIsBinary = strncmp($arrFields[$k], 'binary(', 7) === 0 || strncmp($arrFields[$k], 'blob ', 5) === 0;
```

Comment:
What about `tinyblob`, `mediumblob` and `longblob`? Do we have to handle there here as well?

Reply:
Isn’t this unrelated to this PR?
This pull request only wraps the existing code in an `if (isset($arrFields[$k]))`

Reply:
Ah, "hide whitespace" makes it more clear. Of course then. 👍